### ISSUE TITLE
SSR Modal has client/server hydration mismatch

### ISSUE THREAD

**Issue by chrisvariety** [mantinedev/mantine#22]
Hi there! Trying out this package after seeing it in the [#1 spot on React Status Newsletter](https://react.statuscode.com/issues/237). It looks really well put together so far!

I have a use case that involves rendering a Modal via SSR (not within a Portal, which don't work with SSR anyway). When trying this out there's a client/server hydration mismatch.

Here's a minimal example that stems from your next example, but it will reproduce regardless of SSR framework: https://codesandbox.io/s/condescending-leakey-q3sh2?file=/pages/index.jsx

Just look at the console when loading that page to see an error like:

```
Warning: Prop `id` did not match. Server: "mantine-ayfd64s7a" Client: "mantine-d9vq5oqlf"
    at div
    at u (webpack-internal:///./node_modules/@mantine/core/dist/lib.js:1:5443)
    at div
    at div
    at u (webpack-internal:///./node_modules/@mantine/core/dist/lib.js:1:6852)
    at div
    at div
    at Transition (webpack-internal:///./node_modules/react-transition-group/esm/Transition.js:129:30)
    at s (webpack-internal:///./node_modules/@mantine/core/dist/lib.js:1:108891)
    at O (webpack-internal:///./node_modules/@mantine/core/dist/lib.js:1:139538)
    at Home
    at ThemeProvider (webpack-internal:///./node_modules/theming/dist/theming.esm.js:86:37)
    at c (webpack-internal:///./node_modules/@mantine/core/dist/lib.js:1:46520)
    at App (webpack-internal:///./pages/_app.jsx:29:25)
    at ErrorBoundary (webpack-internal:///./node_modules/@next/react-dev-overlay/lib/internal/ErrorBoundary.js:23:47)
    at ReactDevOverlay (webpack-internal:///./node_modules/@next/react-dev-overlay/lib/internal/ReactDevOverlay.js:73:23)
    at Container (webpack-internal:///./node_modules/next/dist/client/index.js:156:5)
    at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:645:24)
    at Root (webpack-internal:///./node_modules/next/dist/client/index.js:778:24)
```

It seems to be mismatching on the modal title, which is strange. I haven't looked further into it, but happy to help out however I can.

**Comment by rtivital** [comment]
Hi, thanks for feedback and reporting the issue, I actually never tested opened modal in ssr mode, so that's good to know.
The issue here is that modal title and body is [connected to modal](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Modal/Modal.tsx#L118) with aria- attributes via random id, generated by [use-id](https://mantine.dev/hooks/use-id/) hook. 

The problem is that id generates both on client and server and it's **random** in both cases. The only solution I see now is to add `titleId` and `bodyId` props to modal component to ensure they stay the same. I'll fix it in next patch. For now, you have to bare a little with this error message 😄 – this won't impact your application except that.

**Comment by chrisvariety** [comment]
Ah, perfect! Makes sense. Could be a worse problem than that for sure.

**Comment by rtivital** [comment]
[Released in 1.0.4](https://github.com/mantinedev/mantine/releases/tag/1.0.4)
